From 5cb5f11469dfcbd7568fbca8b79c0f20a21cfbf5 Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Fri, 17 Jan 2025 10:09:15 +0100
Subject: [PATCH 2/9] net: airoha: deassert XSI line on hw init

In preparation for phylink support, deassert XSI line as we will naw
make actual use of them for external PHY/SFP cage support.

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 drivers/net/ethernet/airoha/airoha_eth.c | 4 ++++
 1 file changed, 4 insertions(+)

--- a/drivers/net/ethernet/airoha/airoha_eth.c
+++ b/drivers/net/ethernet/airoha/airoha_eth.c
@@ -1383,6 +1383,10 @@ static int airoha_hw_init(struct platfor
 	if (err)
 		return err;
 
+	err = reset_control_bulk_deassert(eth->soc->num_xsi_rsts, eth->xsi_rsts);
+	if (err)
+		return err;
+
 	msleep(20);
 	err = reset_control_bulk_deassert(ARRAY_SIZE(eth->rsts), eth->rsts);
 	if (err)
